zkvm/src/txlog.rs

```diff
@@ -31,6 +32,13 @@ pub struct TxID(pub [u8; 32]);
 #[derive(Copy, Clone, PartialEq, Debug)]
 pub struct UTXO(pub [u8; 32]);
 
+/// MerkleHash represents a step in a Merkle proof of inclusion.
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum MerkleHash {
```

Comment:
Since we'll probably need proofs for multiple items, would it make sense to compute the tree with all intermediate hashes (so we don't repeat them again and again), and then query necessary paths from it at no cost?

Reply:
Bonus points if we can lazily compute/cache/reuse internal hashes, so that if we use the tree for one-two items, we don't pay the cost of hashing all the nodes.